x-pack/legacy/plugins/siem/public/components/open_timeline/delete_timeline_modal/delete_timeline_modal.tsx

```diff
@@ -6,42 +6,79 @@
 
 import { EuiConfirmModal, EUI_MODAL_CONFIRM_BUTTON } from '@elastic/eui';
 import { FormattedMessage } from '@kbn/i18n/react';
-import React from 'react';
+import React, { useCallback, useState } from 'react';
 
 import * as i18n from '../translations';
+import { OpenTimelineResult, SetActionTimeline } from '../types';
 
 interface Props {
-  title?: string | null;
+  title?: string | JSX.Element | null;
   onDelete: () => void;
   closeModal: () => void;
 }
 
 export const DELETE_TIMELINE_MODAL_WIDTH = 600; // px
 
+const getDeletedTitles = (title: string | JSX.Element | null | undefined) => {
+  if (title != null && React.isValidElement(title)) {
+    return title;
+  } else if (title != null && typeof title === 'string' && title.trim().length > 0) {
+    return title.trim();
+  }
+  return `"${i18n.UNTITLED_TIMELINE}"`;
+};
+
 /**
  * Renders a modal that confirms deletion of a timeline
  */
 export const DeleteTimelineModal = React.memo<Props>(({ title, closeModal, onDelete }) => (
   <EuiConfirmModal
+    buttonColor="danger"
+    cancelButtonText={i18n.CANCEL}
+    confirmButtonText={i18n.DELETE}
+    defaultFocusedButton={EUI_MODAL_CONFIRM_BUTTON}
+    onCancel={closeModal}
+    onConfirm={onDelete}
     title={
       <FormattedMessage
         id="xpack.siem.open.timeline.deleteTimelineModalTitle"
         data-test-subj="title"
-        defaultMessage='Delete "{title}"?'
+        defaultMessage="Delete {title}?"
         values={{
-          title: title != null && title.trim().length > 0 ? title.trim() : i18n.UNTITLED_TIMELINE,
+          title: getDeletedTitles(title),
         }}
       />
     }
-    onCancel={closeModal}
-    onConfirm={onDelete}
-    cancelButtonText={i18n.CANCEL}
-    confirmButtonText={i18n.DELETE}
-    buttonColor="danger"
-    defaultFocusedButton={EUI_MODAL_CONFIRM_BUTTON}
   >
     <div data-test-subj="warning">{i18n.DELETE_WARNING}</div>
   </EuiConfirmModal>
 ));
 
 DeleteTimelineModal.displayName = 'DeleteTimelineModal';
+
+export const useDeleteTimeline = ({
+  setActionTimeline,
+}: {
+  setActionTimeline: SetActionTimeline;
+}) => {
+  const [isDeleteTimelineModalOpen, setIsDeleteTimelineModalOpen] = useState<boolean>(false);
+
+  const onCloseDeleteTimelineModal = useCallback(() => {
+    setIsDeleteTimelineModalOpen(false);
+    setActionTimeline(undefined);
+  }, [setIsDeleteTimelineModalOpen]);
```

Comment:
you should have `setActionTimeline` in the dependency since it is coming from the props